app/assets/javascripts/admin/addon/components/ace-editor.js

```diff
@@ -105,6 +105,13 @@ export default Component.extend({
           this.set("content", editor.getSession().getValue());
           this._skipContentChangeEvent = false;
         });
+        editor.commands.addCommand({
+          name: "save",
+          exec: () => {
+            this.attrs.save();
```

Comment:
Maybe this should be something like
```
if(this.attrs.save){
  this.attrs.save()
}
```
So that it doesn't throw an exception if the `save` attribute isn't passed?

Reply:
This was my first thought and it does prevent an error from showing up if the parent component/controller doesn't have a 'save' action. 

However, I'm thinking of putting the entire command declaration within that conditional like this -- 
```
if (this.attrs.save) {
  editor.commands.addCommand({
    name: "save",
    exec: () => {
      this.attrs.save();  
    },
    bindKey: { mac: "cmd-s", win: "ctrl-s" },
  });
}
```
That way, if the parent doesn't define a `save()` action, the browser default save-page dialogue still pops up.